cmd/guacone/cmd/osv.go

```diff
@@ -84,11 +84,55 @@ var osvCmd = &cobra.Command{
 
 		totalNum := 0
 		var totalDocs []*processor.Document
+		docChan := make(chan *processor.Document)
+		tickInterval := 30 * time.Second
+		ticker := time.NewTicker(tickInterval)
 		gotErr := false
+
+		go func() {
+			for !gotErr {
+				select {
+				case <-ticker.C:
+					if len(totalDocs) > 0 {
+						err = ingestor.MergedIngest(ctx, totalDocs, opts.graphqlEndpoint, csubClient)
+						if err != nil {
+							gotErr = true
+							logger.Errorf("unable to ingest documents: %v", err)
+						}
+						totalDocs = []*processor.Document{}
+					}
+					ticker.Reset(tickInterval)
+				case d := <-docChan:
+					totalNum += 1
+					totalDocs = append(totalDocs, d)
+					if len(totalDocs) >= 1000 {
+						err = ingestor.MergedIngest(ctx, totalDocs, opts.graphqlEndpoint, csubClient)
+						if err != nil {
+							gotErr = true
+							logger.Errorf("unable to ingest documents: %v", err)
+						}
+						totalDocs = []*processor.Document{}
+						ticker.Reset(tickInterval)
+					}
+				case <-ctx.Done():
+					return
+				}
+			}
+			for len(docChan) > 0 {
+				totalNum += 1
+				totalDocs = append(totalDocs, <-docChan)
+				if len(totalDocs) >= 1000 {
```

Comment:
if there are any documents left, we should `MergedIngest` them. To capture any stragglers

Reply:
good catch, fixed